### PR TITLE
Run browser visibly

### PR DESCRIPTION
**Pull request by marusak**
When I was working on Firefox CDP I got used to starting browser in !headless mode. So I though we could have it as option.
SO you can run `TEST_SHOW_BROWSER=True ./test/verify/check-menu` and it starts new browser and it is visible. So when you do sit, or even when it fails and you used `-s`, the browser stays opened and you can interact with it.
It needed just one hack, as we first checked for headless chrome executable. And then I switched order in which things are cleaned, so we don't close browser before final sit is running.

And I noticed that I forgot to document TEST_BROWSER so I did that as well.
Running a few tests to see that the browser starts correctly in tasks.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -90,6 +90,11 @@ You can set these environment variables to configure the test suite:
                    the Chrome Debug Protocol, on the given port. Don't use this
                    with parallel tests.
 
+    TEST_BROWSER  What borwser should be used for testing. Currently supported values:
```
Comment (martinpitt) [inline test/README.md]: "borwser"

File: test/README.md [hunk 2]
```diff
@@ -95,6 +95,9 @@ You can set these environment variables to configure the test suite:
                      "firefox"
                   "chromium" is the default.
 
+    TEST_SHOW_BROWSER  Set to `True` to run browser visibly. When not specified,
```
Comment (martinpitt) [inline test/README.md]: `True` is a weird value for shell boolean variables. Usually one just tests non-emptiness.

"visibly" → "interactively" perhaps?

File: test/README.md [hunk 3]
```diff
@@ -95,6 +95,9 @@ You can set these environment variables to configure the test suite:
...
+                       browser is ran in headless mode.
```
Comment (martinpitt) [inline test/README.md]: "is run"

File: test/common/cdp.py [hunk 4]
```diff
@@ -14,9 +14,9 @@
 TEST_DIR = os.path.normpath(os.path.dirname(os.path.realpath(os.path.join(__file__, ".."))))
 
 
-def browser_path(browser):
+def browser_path(browser, show_browser):
     if browser == "chromium":
-        return browser_path_chromium()
+        return browser_path_chromium(show_browser)
     elif browser == "firefox":
         return browser_path_firefox()
```
Comment (martinpitt) [inline test/common/cdp.py]: Showing firefox browser should work as well ideally. Is that just an oversight, or not technically possible right now?

File: test/common/cdp.py [hunk 5]
```diff
@@ -42,14 +42,24 @@ def browser_path_chromium():
...
+            return g[0]
+
+    def standard_chrome():
+        p = subprocess.check_output("which chromium-browser || which chromium || which google-chrome || true",
+                                    shell=True, universal_newlines=True).strip()
+        if p:
+            return p
+
+    browser_types = [standard_chrome, headless_shell]
+    if show_browser is False:
```
Comment (martinpitt) [inline test/common/cdp.py]: `if not show_browser`. Also, why would it pick `headless_shell` for `show_browser`? IMHO it's cleaner to fail with a "not found" error than silently running headless.

File: test/common/cdp.py [hunk 6]
```diff
@@ -71,6 +81,7 @@ def __init__(self, lang=None, verbose=False, trace=False, inject_helpers=[]):
         self.trace = trace
         self.inject_helpers = inject_helpers
         self.browser = os.environ.get("TEST_BROWSER", "chromium")
+        self.show_browser = os.environ.get("TEST_SHOW_BROWSER", False)
```
Comment (martinpitt) [inline test/common/cdp.py]: Let's convert that to a boolean for clarity, i. e. wrap into `bool()`.

File: test/common/cdp.py [hunk 7]
```diff
@@ -167,7 +178,7 @@ def browser_cmd(self, cdp_port, env):
             raise SystemError(self.browser + " is not installed")
 
         if self.browser == "chromium":
-            return [exe, "--headless", "--disable-gpu", "--no-sandbox", "--disable-setuid-sandbox",
+            return [exe, "--headless" if self.show_browser is False else "", "--disable-gpu", "--no-sandbox", "--disable-setuid-sandbox",
```
Comment (martinpitt) [inline test/common/cdp.py]: That looks a bit hackish -- chromium is cool with empty arguments?

File: test/common/testlib.py [hunk 8]
```diff
@@ -793,6 +784,15 @@ def sitter():
...
         self.tmpdir = tempfile.mkdtemp()
 
+        def sitter():
+            if opts.sit and not self.checkSuccess():
+                if self._outcome:
+                    [traceback.print_exception(*e[1]) for e in self._outcome.errors if e[1]]
+                else:
+                    self.currentResult.printErrors()
+                sit(self.machines)
+        self.addCleanup(sitter)
```
Comment (martinpitt) [inline test/common/testlib.py]: Thanks for that fix! This was annoying indeed. Could you please split this out as separate commit?

File: test/common/cdp.py (hunk 4 above)
Comment (marusak) [reply test/common/cdp.py]: Firefox works always. This is the part where we look for browser executable. Every firefox executable we can find supports normal mode. Problem is that we also search for headless chromium, which we don't want to use in this case.

File: test/common/cdp.py (hunk 7 above)
Comment (marusak) [reply test/common/cdp.py]: Tested it and seems fine

File: test/common/cdp.py (hunk 4 above)
Comment (martinpitt) [reply test/common/cdp.py]: Of course, thanks!
